source/dxgi/dxgi.cpp

```diff
@@ -568,9 +31,10 @@ extern "C" HRESULT WINAPI CreateDXGIFactory(REFIID riid, void **ppFactory)
 }
 extern "C" HRESULT WINAPI CreateDXGIFactory1(REFIID riid, void **ppFactory)
 {
-	// Do NOT skip in case this is called internally for D3D10/D3D11 (otherwise the 'IDXGIFactory::CreateSwapChain' call in 'D3D10/11CreateDeviceAndSwapChain' will not be redirected)
+	// External hooks may want to create a DXGI factory and rewrite the vtables
+	if (g_in_dxgi_runtime)
+		return reshade::hooks::call(CreateDXGIFactory1)(riid, ppFactory);
```

Comment:
It looks like this breaks the case where `D3D11CreateDeviceAndSwapChain` is called to create the swapchain, rather than creating only a device and a factory manually and then swapchain from that (note the comment that was here before, I made that mistake before hence putting that comment here =D).

It can probably still be made working by modifying `D3D11CreateDeviceAndSwapChain` to create a temporary instance of `DXGIFactory` with the factory object it queries and call `CreateSwapChain` on that (https://github.com/crosire/reshade/blob/main/source/d3d11/d3d11.cpp#L149).

Same thing also applies to `D3D10CreateDeviceAndSwapChain`.